draw/src/main/java/org/odk/collect/draw/DrawViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package org.odk.collect.draw
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import org.odk.collect.async.Scheduler
+import timber.log.Timber
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+
+internal class DrawViewModel(
+    private val output: File,
+    private val scheduler: Scheduler
+) : ViewModel() {
+
+    private val _saveResult = MutableLiveData<Boolean?>(null)
```

Comment:
Because it doesn't get set until there has been a save meaning we need to differentiate between "save failure" (`false`) and "hasn't saved yet" (`null`).

Reply:
Can't no value represent that `hasn't saved yet` state?

Reply:
Oh yeah I'd forgotten about that change where we get proper nullability! Bear with me, I'd got so used to having to have null checks in all observers regardless of the type.